middleware/faq.rst

```diff
@@ -541,6 +541,35 @@ In most cases, a complex ``--data-query`` argument is preferable to even one inp
+The primary tools here are the ``--skip-existing-in`` and ``--select-tasks`` options to ``pipetask qgraph`` (or equivalently, the BPS ``extraQgraphArgs`` setting).
+The former is given a collection that is queried for existing outputs, and any quanta that already have a metadata dataset present in that collection are considered done and dropped from the new graph.
+The latter takes a :ref:`pipeline graph subset expression <pipeline-graph-subset-expressions>` that can be used to easily identify the tasks that were affected by a failure.
+
+There are a few different scenarios to consider:
+
+- If a task with label ``unfixable`` fails in one or more quanta, and you just want to move on without its outputs, modify the original quantum-graph build command with ``--skip-existing-in <output> --select-tasks '> unfixable'``.
+  This will only attempt to run tasks that are strictly downstream of ``unfixable``, and it will skip any quanta that already ran (i.e. because they didn't have upstream failures).
+
+- If a task with label ``fixable`` fails in one more quanta, but you have fixed the problem and now expect it to succeed at least some of the time, use ``--skip-existing-in <output> --select-tasks '>= fixable'``.
+  This will run both the problematic task and everything downstream of it, again skipping any quanta that already ran successfully.
+  In this case ``-skip-existing-in <output>`` is actually all you need for correctness, but using ``--select-tasks`` should speed up the quantum graph build.
+
+- If a task with label ``regrettable`` doesn't actually fail, but it's produced bad outputs (or an improvement is available that you want to try), use ``--select-tasks '>= regrettable'`` to re-run that task and all downstream tasks regardless of whether they succeeded before.
+  This is also the right approach to take with quanta that incorrectly raised `NoWorkFound` or otherwise wrote no outputs without formally failing and blocking downstream quanta; these are considered "successes with caveats", not failures.
+
+When the data IDs of the problems are known in any of these cases, it may be possible to further limit the set of quanta re-run using the ``--data-query`` and/or ``--data-id-table`` options.
+These data ID constraints act on *all* downstream quanta (often via spatial joins), not just the failures, however, so it's safest to specify full ``visits`` or ``tracts`` to avoid "region slicing" problems (in which e.g. a downstream ``tract``-level task that needs full ``visits`` isn't recognized as overlapping a particular ``{visit, detector}`` that doesn't overlap the ``tract``).
```

Comment:
I had to read this multiple times.   I don't know if breaking it up would help or not.  Part of the confusion was trying to figure out the example (the overlapping, but doesn't overlap), but I suspect this example is more familiar to the typical audience of this FAQ.

Reply:
I've rewritten this entirely with a more complete example.